zerver/models.py

```diff
@@ -1814,6 +1814,13 @@ class UserProfile(AbstractBaseUser, PermissionsMixin, UserBaseSettings):  # type
     full_name = models.CharField(max_length=MAX_NAME_LENGTH)
 
     date_joined = models.DateTimeField(default=timezone_now)
+
+    # Terms of Service version number that this user has accepted. We
+    # use the special value TOS_VERSION_BEFORE_FIRST_LOGIN for users
+    # whose account was created without direct user interaction (via
+    # the API or a data import), and null for users whose account is
+    # fully created on servers that do not have a configured ToS.
```

Comment:
Added this block comment, since this is starting to deserve some explanation :).